colossalai/engine/paramhooks/_base_paramhook.py

```diff
@@ -0,0 +1,37 @@
+from typing import Callable, List
+import torch
+
+class BaseParamHook(object):
+    def __init__(self, param_list: List[torch.nn.Parameter], backward_hook_call: Callable) -> None:
+        r"""
+        register backward hook on every parameters of module
+        """
+        self.param_list = param_list
+        self._register_backward_hooks(backward_hook_call)
+
+    def _register_backward_hooks(self, hook_call : Callable) -> None:
+        r"""
+        The hook_call will be called every time a gradient with respect to the a param in self.param_list 
+        is computed. 
+        The hook should have the following signature:
+        ```
+        hook(grad) -> Tensor or None
+        ```
+        """
+        if not torch.is_grad_enabled():
+            return  # don't register grad hooks if grad isn't enabled
+        for p in self.param_list:
+            if p.requires_grad and not hasattr(p, 'zero_shard_bwd_hook'):
```

Comment:
Hi, I think we should not use 'zero_shard_bwd_hook' as the attribute name. Maybe we should use a more general one.